### PR TITLE
OCP 3.5 Release Notes

### PR DESCRIPTION
**Pull request by ahardin-rh**
Preview Build:
http://file.rdu.redhat.com/~ahardin/03072017/OCP-3.5-release-notes/release_notes/ocp_3_5_release_notes.html

### CODE REVIEW

**Comment by adellape** [comment]
@ahardin-rh This is the OVS card I was mentioning: https://trello.com/c/2XRQgOIs

Basically we can just say under `[[ocp-35-updated-infrastructure-components]]` that OVS was upgraded to 2.6 and the package is now provided via the Red Hat Enterprise Linux Fast Datapath channel.

**Comment by ahardin-rh** [comment]
@adellape @aheslin @vikram-redhat @mpbarrett This is my first pass at the OCP 3.5 release notes. Please review. Thank you!

Preview Build: http://file.rdu.redhat.com/~ahardin/03072017/OCP-3.5-release-notes/release_notes/ocp_3_5_release_notes.html

File: release_notes/ocp_3_5_release_notes.adoc [hunk 1]
```diff
@@ -0,0 +1,1683 @@
...
+* Dates now use the word form over the number form to avoid ambiguity (For example,
+May 4, 2016 instead of 05/04/2017).
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1333101[*BZ#1333101*])
+
+* Previously, dropdown menus on the web console overlay the navigation menu
+dropdowns, blocking the view and usability of the navigation menu dropdowns. The
+navigation menu dropdown’s z-index has been set to a value greater than that of
+page content dropdowns, resulting in navigation menu dropdowns to always appear
+on top of page content dropdowns.
+(l
```
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: @mpbarrett Is there anything that should be listed in this section? I had nothing in my notes for 3.5. Thanks!

Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: @mpbarrett Is there anything that should be listed in the Known Issues section? I did not have anything in my notes. Thanks!

File: release_notes/ocp_3_5_release_notes.adoc [hunk 2]
```diff
@@ -0,0 +1,1683 @@
...
+link:https://bugzilla.redhat.com/show_bug.cgi?id=1421834[(BZ#1421834)].
+
+Increasing the `METRICS_RESOLUTION` interval helped achieve better results in
+relation to how many pods can be monitored by one set of metrics pods. In
+{product-title} 3.5, tests showed that OpenShift metrics collection was stable for
+test cases up to 25,000 monitored pods in a {product-title} cluster.
+
+See
+xref:../install_config/cluster_metrics.adoc#metrics-recommendations-for-OCP-version-35[Enabling
+Cluster Metrics] for more information.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: This link will change once https://github.com/openshift/openshift-docs/pull/3881 lands

File: release_notes/ocp_3_5_release_notes.adoc [hunk 3]
```diff
@@ -0,0 +1,1683 @@
...
+
+See more information about xref:ocp-35-web-console-statefulsets[web console
+enhancements] related to this feature for {product-title} 3.5.
+
+[[ocp-35-registry]]
+=== Registry
+
+{product-title} now allows control of whether or not an image is cached locally
+in the internal OpenShift Registry via the `oc tag` command with the
+`--reference-policy=local` and ``--scheduled=true` options.
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: Double backticks here.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 4]
```diff
@@ -0,0 +1,1683 @@
...
+ordinality changes or mounted PVs.
+
+See more information about xref:ocp-35-web-console-statefulsets[web console
+enhancements] related to this feature for {product-title} 3.5.
+
+[[ocp-35-registry]]
+=== Registry
+
+{product-title} now allows control of whether or not an image is cached locally
+in the internal OpenShift Registry via the `oc tag` command with the
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: Should we change all references to be "OpenShift Container Registry"?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 5]
```diff
@@ -0,0 +1,1683 @@
...
+`--reference-policy=local` and ``--scheduled=true` options.
+
+The storage of the manifest is moved to the OpenShift Registry, instead of
+storing it in etcd. There are two processes that will clean up existing images'
+metadata from etcd:
+
+* Gradually, `push` and `prune` will migrate all etcd images to not have manifest attached.
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: s/have/have the/

File: release_notes/ocp_3_5_release_notes.adoc [hunk 6]
```diff
@@ -0,0 +1,1683 @@
...
+regeneration behavior.
+
+See xref:../dev_guide/secrets.adoc#service-serving-certificate-secrets[Service Serving Certificate Secrets] for more information.
+
+[[ocp-35-configurable-expiry-range-for-framework-certs]]
+==== Configurable Expiry Range for Framework Certificates
+
+By default, the certificates used to govern the etcd, master, and kubelet expire
+after two to five years. There is now an `oc` command to change this expiry to
+be end-user configurable. This has not been implemented in Ansible yet.
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: Probably means to more specifically say "in the Ansible installer yet"?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 7]
```diff
@@ -0,0 +1,1683 @@
...
+$ oc policy can-i --list --user=**
+$ oc policy can-i --list --groups=**
+----
+
+List which permissions a particular user or group has in the project by system
+administrator role:
+
+----
+$ oc policy can-i --list --user=** -n project
+$ oc policy can-i --list --groups=**  -n project
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]:
```
<project>
```

File: release_notes/ocp_3_5_release_notes.adoc [hunk 8]
```diff
@@ -0,0 +1,1683 @@
...
+----
+
+Determine if users can have all the combination of verbs and resources from `oc
+policy can-i --list --user/group`
+
+----
+$ oc policy can-i <verb> <resource> --user/group
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: I think the "group" is actually "--groups" (plural). From oc 3.5:

http://pastebin.test.redhat.com/473925

How bout for all of the "--user/group", instead have:

```
[--user|--groups]
```
Showing that it's optional, and that you have a choice between the two?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 9]
```diff
@@ -0,0 +1,1683 @@
...
+Test the SCCs with scopes: `oc policy can-i --user/group`
+
+----
+$ oc policy can-i verb resource --user/group --scopes=user:info
+$ oc policy can-i verb resource --user/group --scopes=user:info,role:admin:<namespace>
+$ oc policy can-i verb resource --user/group --scopes=role:view:*
+$ oc policy can-i verb resource --user/group --scopes=role:edit:*
+$ oc policy can-i verb resource --user/group --scopes=role:admin:*
+$ oc policy can-i verb resource --user/group --scopes=role:admin:*:!
+----
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: All of the "verb resource" could be `<verb> <resource>` like you had in the first example.

**Review by adellape — COMMENTED** [review]
@ahardin-rh Adding the comments I have so far.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 10]
```diff
@@ -0,0 +1,1683 @@
...
+new resource asynchronously by the kubelet.
+
+[[ocp-35-statefulsets]]
+==== StatefulSets (Technology Preview)
+
+`StatefulSets` (currently in xref:ocp-35-technology-preview[Technology Preview]
+and formerly known as `PetSets`) offer more control over scale, network naming,
+handling of PVs, and deployment sequencing.
+
+This new controller allows for the deployment of applications types that require
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: s/applications/application

File: release_notes/ocp_3_5_release_notes.adoc (hunk 5 above)
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Serial comma after "push".

File: release_notes/ocp_3_5_release_notes.adoc [hunk 11]
```diff
@@ -0,0 +1,1683 @@
...
+----
+
+See
+xref:../architecture/additional_concepts/authorization.adoc#authorization-determining-what-you-can-do-as-an-authenticated-user[Authorization]
+for more information.
+
+[[ocp-35-github-identity-provider-can-optionally-require-a-team]]
+==== GitHub Identity Provider Can Optionally Require a Team
+
+Users can now test for GitHub team membership at login.
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: s/login/log in

File: release_notes/ocp_3_5_release_notes.adoc [hunk 12]
```diff
@@ -0,0 +1,1683 @@
...
+
+----
+$ oc create route edge edgeroute --service=service-secure --wildcard-policy=Subdomain --hostname=edge.edgeroute.com
+----
+
+Test the route:
+
+----
+$ curl --resolve edge2.edgeroute.com:443:$router_ip https://edge2.edgeroute.com -k
+----
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Should all of the above be in steps?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 13]
```diff
@@ -0,0 +1,1683 @@
...
+
+.Add Config Files
+image::ocp-35-add-config-files.png[Add Config Files]
+
+
+[[ocp-35-web-console-show-build-failures]]
+==== Show Build Failures
+
+Users no longer have to search logs to gain a better understanding of why build
+failed. Individual build status messages  are now updated with details that are
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Double space after "messages".

File: release_notes/ocp_3_5_release_notes.adoc [hunk 14]
```diff
@@ -0,0 +1,1683 @@
...
+[WARNING]
+====
+When `ingressIPNetworkCIDR` changes, any external IPs allocated from the
+previous range will be reallocated from the new range.
+====
+
+* The `groups` field in the user object is now deprecated. Instead, create Group
+API objects containing the names of the users that are members of the group.
+
+* `oc whoami --token` was deprecated in 1.4 in favor of `oc whoami -t`. Also, `oc
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Should 1.4 above be {product-title} 3.4?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 15]
```diff
@@ -0,0 +1,1683 @@
...
+now has proper indentation and colorized output.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1397995[*BZ#1397995*])
+
+* Output from the `oc idle` command was confusing to end users. A user could not
+easily tell what was being done by the `oc idle` command. With this bug fix, the
+output of the `oc idle` command was updated to clarify what the command had done
+and is now easier to understand.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1402356[*BZ#1402356*])
+
+* Previously,  `oc status` tried to generate a status for the "default" cluster
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Double space above.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 16]
```diff
@@ -0,0 +1,1683 @@
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1393943[*BZ#1393943*])
+
+* There was a duplicated resource "quota" in the `oc describe` list of valid
+resources. Therefore, "quota" was printed twice. This bug fix removes one entry
+on "quota" in the `oc describe` list of valid resources. Now, each resource type
+is only printed once.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1396397[*BZ#1396397*])
+
+* Multi-line output for a template description did not display all lines with
+ correct indentation under `oc new-app` . Therefore, the output for template
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Space after oc-new-app

File: release_notes/ocp_3_5_release_notes.adoc [hunk 17]
```diff
@@ -0,0 +1,1683 @@
...
+
+* The *Browse* tab now shows the local host name of a service.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1395821[*BZ#1395821*]
+
+* On a project’s *Settings* tab, the Quota terminating scope descriptions are not
+clear
+
+* The spinning icon on the *Browse* tab that indicates a pod is running no longer
+appears jittery. In some browser/operating system combinations, font and
+line-height issues could make a spinning icon wobble.  Those issues have been
```
Comment (bfallonf) [inline release_notes/ocp_3_5_release_notes.adoc]: Double space above

**Comment by bfallonf** [comment]
I made some comments. Just typo-level stuff. Good job overall :1st_place_medal:

File: release_notes/ocp_3_5_release_notes.adoc [hunk 18]
```diff
@@ -0,0 +1,1683 @@
...
+
+- Slow to iterate through the Ordinal Index and, therefore, slow on scale up and
+scale down.
+- No deployment or pod specification post deployment verification of what is
+deployed versus what is configured in the JSON file.
+- Locality awareness of zones or regions when dealing with scale up or scale down
+ordinality changes or mounted PVs.
+
+See more information about xref:ocp-35-web-console-statefulsets[web console
+enhancements] related to this feature for {product-title} 3.5.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: Good information. Thank you!

File: release_notes/ocp_3_5_release_notes.adoc (hunk 5 above)
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: @bfallonf "Gradually" should not be part of the list. I am trying to convey that the etcd images are being gradually migrated. I reworded this to add clarity.

File: release_notes/ocp_3_5_release_notes.adoc (hunk 12 above)
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: I considered it, but the commands are not necessarily sequential.

File: release_notes/ocp_3_5_release_notes.adoc (hunk 14 above)
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: Yes, good catch :baseball:

**Comment by ahardin-rh** [comment]
@adellape @bfallonf Thank you! Your feedback is now incorporated.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 19]
```diff
@@ -0,0 +1,1691 @@
...
+
+Built on Red Hat Enterprise Linux and Google Kubernetes, {product-title}
+provides a secure and scalable multi-tenant operating system for today’s
+enterprise-class applications, while providing integrated application runtimes
+and libraries. {product-title} brings the OpenShift PaaS platform to customer
+data centers, enabling organizations to implement a private PaaS that meets
+security, privacy, compliance, and governance requirements.
+
+[[ocp-34-about-this-release]]
+== About This Release
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: s/34/35/

File: release_notes/ocp_3_5_release_notes.adoc [hunk 20]
```diff
@@ -0,0 +1,1691 @@
...
+xref:../install_config/index.adoc#install-config-index[Installation and
+Configuration] documentation.
+
+To upgrade to this release from a previous version, see the
+xref:../install_config/upgrading/index.adoc#install-config-upgrading-index[Upgrading
+a Cluster] topics in the
+xref:../install_config/index.adoc#install-config-index[Installation and
+Configuration] documentation.
+
+[[ocp-34-new-features-and-enhancements]]
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: s/34/35/

File: release_notes/ocp_3_5_release_notes.adoc [hunk 21]
```diff
@@ -0,0 +1,1691 @@
...
+[[ocp-35-platform-management]]
+=== Platform Management
+
+[[ocp-35-application-service-cert-regeneration]]
+==== Application Service Certificate Regeneration (Technology Preview)
+
+Application service certificate regeneration is currently in
+xref:ocp-35-technology-preview[Technology Preview].
+
+The controller will now look over the expiry of application CERTs that have used
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: s/CERTs/certificates/ ?

File: release_notes/ocp_3_5_release_notes.adoc [hunk 22]
```diff
@@ -0,0 +1,1691 @@
...
+correctly validate these fields so that you cannot submit the form with invalid
+values. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1419887[*BZ#1419887*])
+
+* Previously, the link to download the OpenShift CLI linked to the incorrect
+version (of OpenShift Origin). The link has been updated, and the link downloads
+the correct version.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1421949[*BZ#1421949*])
+
+* Previously, the *Create a Secret* and *Add Config Files* buttons when creating
+using the web console mistakenly linked to the other page. The buttons have bee
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: s/CERT/Certificate/ ?

Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: I'd kill this RFE section and just move BZ#1317159 under Routing.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 23]
```diff
@@ -0,0 +1,1691 @@
...
+severe than it is.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1398330[*BZ#1398330*])
+
+*Command Line Interface*
+
+* The latest version of Docker for Mac/Windows uses the Community Edition
+versioning scheme. This causes 'oc cluster up' to halt with an error because
+the new version cannot be parsed by the 'semver' library. This bug fix
+changes the behavior to display a warning instead of exiting with an
+Error. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1428978[*BZ#1428978*])
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: Change quotes to backticks.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 24]
```diff
@@ -0,0 +1,1691 @@
...
+
+{product-title} 3.5 introduces the following notable technical changes.
+
+[discrete]
+[[ocp-35-updated-infrastructure-components]]
+=== Updated Infrastructure Components
+
+{product-title} 3.5 is supported on RHEL 7.2 and 7.3 with the latest packages
+from Extras, including Docker 1.12. {product-title} 3.5 is not certified with
+older versions of Docker and RHEL 7.1 or earlier.
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: We should mention Kubernetes has been updated to v1.5 and etcd to just 3.1. A la:

https://docs.openshift.com/container-platform/3.4/release_notes/ocp_3_4_release_notes.html#ocp-34-notable-technical-changes

**Review by adellape — COMMENTED** [review]
Some more nits.

File: release_notes/ocp_3_5_release_notes.adoc [hunk 25]
```diff
@@ -0,0 +1,1691 @@
...
+3.5 cluster. The storage prefix has changed, along with the name, and newly
+created clusters do not know where to look for `ScheduledJob`. Cluster version
+3.5 operates on `CronJob`, but it also understands `ScheduledJob` submitted to
+it. It performs rapid conversion, saving your newly created object as a
+`CronJob`, resulting in all subsequent read operations returning `CronJob`
+instead. See xref:../dev_guide/cron_jobs.adoc#dev-guide-cron-jobs[Cron Jobs] for
+more information.
+
+* The default value for `ingressIPNetworkCIDR` was previously a non-private range
+(`172.46.0.0/16`) and has been changed to a private range ``(172.29.0.0/16`).
```
Comment (adellape) [inline release_notes/ocp_3_5_release_notes.adoc]: Extra backtick here.

File: release_notes/ocp_3_5_release_notes.adoc (hunk 22 above)
Comment (ahardin-rh) [inline release_notes/ocp_3_5_release_notes.adoc]: Found a few other instances. Updated.

**Pull request merged**